javaslang/src/main/java/javaslang/collection/Tree.java

```diff
@@ -748,12 +771,12 @@ public boolean isEmpty() {
 
         @Override
         public boolean isLeaf() {
-            return children.isEmpty();
+            return size == 1;
         }
 
         @Override
         public int size() {
-            return 1 + children.foldLeft(0, (acc, child) -> acc + child.length());
+            return size;
```

Comment:
@paplorinc @ruslansennov ok, for trees with a huge branching factor it may be worth saving the time - let's experiment with it. I'm mainly interested in saving the memory. Ruslan, the 48 bytes are just for the reference, not for the whole Lazy object, aren't they? We also have a Supplier within the Lazy etc.